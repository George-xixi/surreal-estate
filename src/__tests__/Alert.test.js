import React from "react";
import { render, screen } from "@testing-library/react";
import Alert from "../components/Alert";

describe("Alert", () => {
  it("displays error message correctly", () => {
    const { asFragment } = render(<Alert message="Error!" />);
    const errorElement = screen.getByText("Error!");

    expect(asFragment()).toMatchSnapshot();
    expect(errorElement).toBeInTheDocument();
  });

  it("displays success message correctly", () => {
    const { getByText, asFragment } = render(
      <Alert message="Success!" success />
    );

    expect(asFragment()).toMatchSnapshot();
    expect(getByText(/Success/).textContent).toBe("Success!");
  });

  it("does not render anything if there is no message prop", () => {
    const { asFragment } = render(<Alert message="" />);
    const alert = asFragment();

    expect(alert).toMatchSnapshot();
  });
});
