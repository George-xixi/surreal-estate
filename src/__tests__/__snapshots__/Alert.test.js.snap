// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert displays error message correctly 1`] = `
<DocumentFragment>
  <div
    class="alert alert-error"
  >
    Error!
  </div>
</DocumentFragment>
`;

exports[`Alert displays success message correctly 1`] = `
<DocumentFragment>
  <div
    class="alert alert-success"
  >
    Success!
  </div>
</DocumentFragment>
`;

exports[`Alert does not render anything if there is no message prop 1`] = `<DocumentFragment />`;
